### ISSUE TITLE
demo, multi region: 2x slowdown in making movr database multi region

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#100894]
Running on current master with
```
./cockroach demo movr --global --nodes 9 --multitenant=false --insecure
```
this whole snippet to convert the `movr` database into MR (copy-paste into a file, due to #100892):
<details>

```sql
ALTER DATABASE movr PRIMARY REGION "us-east1";
ALTER DATABASE movr ADD REGION "europe-west1";
ALTER DATABASE movr ADD REGION "us-west1";

ALTER TABLE promo_codes SET LOCALITY GLOBAL;

ALTER TABLE rides ADD COLUMN region crdb_internal_region AS (
  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
       WHEN city = 'paris' THEN 'europe-west1'
       WHEN city = 'rome' THEN 'europe-west1'
       WHEN city = 'new york' THEN 'us-east1'
       WHEN city = 'boston' THEN 'us-east1'
       WHEN city = 'washington dc' THEN 'us-east1'
       WHEN city = 'san francisco' THEN 'us-west1'
       WHEN city = 'seattle' THEN 'us-west1'
       WHEN city = 'los angeles' THEN 'us-west1'
  END
) STORED;
ALTER TABLE rides ALTER COLUMN region SET NOT NULL;
ALTER TABLE rides SET LOCALITY REGIONAL BY ROW AS "region";

ALTER TABLE user_promo_codes
  ADD CONSTRAINT user_promo_codes_code_fk
    FOREIGN KEY (code)
    REFERENCES promo_codes (code)
    ON UPDATE CASCADE;
ALTER TABLE user_promo_codes ADD COLUMN region crdb_internal_region AS (
  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
       WHEN city = 'paris' THEN 'europe-west1'
       WHEN city = 'rome' THEN 'europe-west1'
       WHEN city = 'new york' THEN 'us-east1'
       WHEN city = 'boston' THEN 'us-east1'
       WHEN city = 'washington dc' THEN 'us-east1'
       WHEN city = 'san francisco' THEN 'us-west1'
       WHEN city = 'seattle' THEN 'us-west1'
       WHEN city = 'los angeles' THEN 'us-west1'
  END
) STORED;
ALTER TABLE user_promo_codes ALTER COLUMN region SET NOT NULL;
ALTER TABLE user_promo_codes SET LOCALITY REGIONAL BY ROW AS "region";

ALTER TABLE users ADD COLUMN region crdb_internal_region AS (
  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
       WHEN city = 'paris' THEN 'europe-west1'
       WHEN city = 'rome' THEN 'europe-west1'
       WHEN city = 'new york' THEN 'us-east1'
       WHEN city = 'boston' THEN 'us-east1'
       WHEN city = 'washington dc' THEN 'us-east1'
       WHEN city = 'san francisco' THEN 'us-west1'
       WHEN city = 'seattle' THEN 'us-west1'
       WHEN city = 'los angeles' THEN 'us-west1'
  END
) STORED;
ALTER TABLE users ALTER COLUMN region SET NOT NULL;
ALTER TABLE users SET LOCALITY REGIONAL BY ROW AS "region";

ALTER TABLE vehicle_location_histories ADD COLUMN region crdb_internal_region AS (
  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
       WHEN city = 'paris' THEN 'europe-west1'
       WHEN city = 'rome' THEN 'europe-west1'
       WHEN city = 'new york' THEN 'us-east1'
       WHEN city = 'boston' THEN 'us-east1'
       WHEN city = 'washington dc' THEN 'us-east1'
       WHEN city = 'san francisco' THEN 'us-west1'
       WHEN city = 'seattle' THEN 'us-west1'
       WHEN city = 'los angeles' THEN 'us-west1'
  END
) STORED;
ALTER TABLE vehicle_location_histories ALTER COLUMN region SET NOT NULL;
ALTER TABLE vehicle_location_histories SET LOCALITY REGIONAL BY ROW AS "region";

ALTER TABLE vehicles ADD COLUMN region crdb_internal_region AS (
  CASE WHEN city = 'amsterdam' THEN 'europe-west1'
       WHEN city = 'paris' THEN 'europe-west1'
       WHEN city = 'rome' THEN 'europe-west1'
       WHEN city = 'new york' THEN 'us-east1'
       WHEN city = 'boston' THEN 'us-east1'
       WHEN city = 'washington dc' THEN 'us-east1'
       WHEN city = 'san francisco' THEN 'us-west1'
       WHEN city = 'seattle' THEN 'us-west1'
       WHEN city = 'los angeles' THEN 'us-west1'
  END
) STORED;
ALTER TABLE vehicles ALTER COLUMN region SET NOT NULL;
ALTER TABLE vehicles SET LOCALITY REGIONAL BY ROW AS "region";

SELECT '                                       done';
```

</details>

Takes 3m 40s on my laptop whereas on 22.2.6 it takes 1m 45s, so it's about 2x slow down.

Jira issue: CRDB-26677

**Comment by rafiss** [comment]
I wonder if this could be related to any of the MR Serverless work. cc @JeffSwenson for awareness.

**Comment by JeffSwenson** [comment]
I don't think the MR Serverless changes should have any impact on this. They only kick in if the system database is configured as multi-region, which should not be happening here.

**Comment by rafiss** [comment]
cc @chengxiong-ruan would you mind trying this and profiling it to see if we have any obvious slowdowns?

**Comment by chengxiong-ruan** [comment]
I've tried to repro the slowness. But I experienced much inconsistency from run to run without setting the lease preference of system database. The way I repro is doing `\i script.sql` after the demo started up with `./cockroach demo movr --global --nodes 9 --multitenant=false --insecure`.

After setting `ALTER DATABASE system CONFIGURE ZONE USING constraints = '{+region=us-east1: 1}', lease_preferences = '[[+region=us-east1]]';` I got more consistent execution time from both `22.2.6` and `23.1.0-rc.2`. And they're close around `100s` v.s `110s`. Execution time varies a bit from time to time, but I never seen it doubled.

@yuzefovich let me know if you're happy with this.
